pkg/utils/utils.go

```diff
@@ -277,26 +271,39 @@ func CustomizeUIRouteSpec(routeSpec *routev1.RouteSpec,
 // CustomizeDeployment ...
 func CustomizeDeployment(deploy *appsv1.Deployment, instance *kappnavv1.Kappnav) {
 	deploy.Labels = GetLabels(instance, deploy.Labels, &deploy.ObjectMeta, "")
+	
 	// Ensure that there's at least one replica
 	if deploy.Spec.Replicas == nil || *deploy.Spec.Replicas < 1 {
 		one := int32(1)
 		deploy.Spec.Replicas = &one
 	}
+	
+	// Add label selector 
 	deploy.Spec.Selector = &metav1.LabelSelector{
 		MatchLabels: map[string]string{
 			"app.kubernetes.io/component": deploy.GetName(),
 		},
 	}
+	
 }
 
 // CustomizePodSpec ...
 func CustomizePodSpec(pts *corev1.PodTemplateSpec, parentComponent *metav1.ObjectMeta,
 	containers []corev1.Container, volumes []corev1.Volume, instance *kappnavv1.Kappnav) {
+	
 	pts.Labels = GetLabels(instance, pts.Labels, parentComponent, "")
-	pts.Spec.Containers = containers
-	pts.Spec.RestartPolicy = corev1.RestartPolicyAlways
-	pts.Spec.ServiceAccountName = instance.GetName() + "-" + ServiceAccountNameSuffix
-	pts.Spec.Volumes = volumes
+	if pts.Spec.Containers == nil {
+		pts.Spec.Containers = containers
```

Comment:
The operator's job is to keep the resources it creates/maintains in the desired state. This lets others replace all of our containers with arbitrary ones that may not even be from kAppNav. We're not managing the deployment correctly if we allow this. I think this would also miss updates to config from the kAppNav CR, for instance memory/CPU restrictions that could be updated any time.

Reply:
I see ui and controller deployment are repeatedly updated (reconciled: status:  updated) and it looks like it just get same information on each update.  If it is working as design, I can leave it unchanged.

Reply:
I'm not saying that it's working as designed. I haven't investigated to determine whether that's true or not, but I am saying that this change has undesirable side effects. I think we need to determine what is actually changing in the deployment and make the operator more tolerant so that it doesn't get stuck in an infinite loop of Reconcile() calls.